templates/CRM/Contribute/Form/ContributionPage/Widget.tpl

```diff
@@ -8,122 +8,73 @@
  +--------------------------------------------------------------------+
 *}
 {crmRegion name="contribute-form-contributionpage-widget-main"}
-<h3>{ts}Configure Widget{/ts}</h3>
-{if $showStatus}
-<div class="messages status no-popup">
-    {icon icon="fa-info-circle"}{/icon}
-    {ts}It looks like you may have posted and / or distributed the flash version of the Contribution widget. We won't be supporting the flash version in next release. You should try and get all sites using the flash widget to update to the improved HTML widget code below as soon as possible.{/ts}
+<div class="help">
+  {ts}CiviContribute widgets allow you and your supporters to easily promote this fund-raising campaign. Widget code can be added to any web page.  It will provide a real-time display of current contribution results and a direct link to this contribution page.{/ts} {help id="id-intro"}
 </div>
-{/if}
 <div id="form" class="crm-block crm-form-block crm-contribution-contributionpage-widget-form-block">
-    <div class="help">
-        {ts}CiviContribute widgets allow you and your supporters to easily promote this fund-raising campaign. Widget code can be added to any web page.  It will provide a real-time display of current contribution results and a direct link to this contribution page.{/ts} {help id="id-intro"}
-    </div>
+  <div id="preview" class="float-right">
+    {if $widget_id}
+      <div>
+        {include file="CRM/Contribute/Page/Widget.tpl" widgetId=$widget_id cpageId=$cpageId}<br />
+      </div>
+      <div>
+        <div class="description">{ts}Add this widget to any web page by copying and pasting the code below.{/ts}</div>
+        <textarea rows="8" cols="50" name="widget_code" id="widget_code">{include file="CRM/Contribute/Page/Widget.tpl" widgetId=$widget_id cpageId=$cpageId}</textarea>
+        <br />
+        <strong><a href="#" onclick="Widget.widget_code.select(); return false;"><i class="crm-i fa-code" aria-hidden="true"></i> {ts}Select Code{/ts}</a></strong>
+      </div>
+    {else}
+      <div class="description">{ts}The code for adding this widget to web pages will be displayed here.{/ts}</div>
+    {/if}
+  </div>
+  <table class="form-layout-compressed">
+    <tr class="crm-contribution-contributionpage-widget-form-block-is_active"><td style="width: 12em;">&nbsp;</td><td style="font-size: 10pt;">{$form.is_active.html}&nbsp;{$form.is_active.label}</td></tr>
+  </table>
+  <div class="spacer"></div>
+  <div id="widgetFields">
     <table class="form-layout-compressed">
-      <tr class="crm-contribution-contributionpage-widget-form-block-is_active"><td style="width: 12em;">&nbsp;</td><td style="font-size: 10pt;">{$form.is_active.html}&nbsp;{$form.is_active.label}</td></tr>
+      <tr class="crm-contribution-contributionpage-widget-form-block-title">
+        <td class="label">{$form.title.label}<span class="crm-marker"> *</span></td>
```

Comment:
Does this need HTML escaping?

Reply:
Probably yes

Reply:
hmm I don't understand. Why escape this, and not other labels?

The rendering of the widget itself uses JSON, it does not display any user input directly.

Reply:
Oh, sorry, a quickform label includes the full markup like `<label>...</label>`, so this should *not* be escaped.